jdeeco-core/src/cz/cuni/mff/d3s/deeco/runtime/DEECo.java

```diff
@@ -0,0 +1,226 @@
+package cz.cuni.mff.d3s.deeco.runtime;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Set;
+
+import cz.cuni.mff.d3s.deeco.annotations.processor.AnnotationProcessor;
+import cz.cuni.mff.d3s.deeco.annotations.processor.AnnotationProcessorException;
+import cz.cuni.mff.d3s.deeco.executor.Executor;
+import cz.cuni.mff.d3s.deeco.executor.SameThreadExecutor;
+import cz.cuni.mff.d3s.deeco.knowledge.CloningKnowledgeManagerFactory;
+import cz.cuni.mff.d3s.deeco.knowledge.KnowledgeManagerContainer;
+import cz.cuni.mff.d3s.deeco.knowledge.KnowledgeManagerFactory;
+import cz.cuni.mff.d3s.deeco.model.runtime.api.ComponentInstance;
+import cz.cuni.mff.d3s.deeco.model.runtime.api.EnsembleDefinition;
+import cz.cuni.mff.d3s.deeco.model.runtime.api.RuntimeMetadata;
+import cz.cuni.mff.d3s.deeco.model.runtime.custom.RuntimeMetadataFactoryExt;
+import cz.cuni.mff.d3s.deeco.scheduler.Scheduler;
+import cz.cuni.mff.d3s.deeco.scheduler.SingleThreadedScheduler;
+
+/**
+ * Main container for a DEECo application.
+ * 
+ * @author Ilias Gerostathopoulos <iliasg@d3s.mff.cuni.cz>
+ * @author Filip Krijt <krijt@d3s.mff.cuni.cz>
+ */
+public class DEECo implements DEECoContainer {
+
+	/**
+	 * The metadata model corresponding to the running application.
+	 */
+	RuntimeMetadata model;
+	
+	/**
+	 * The core plugin of the architecture.
+	 */
+	RuntimeFramework runtime;
+	
+	/**
+	 * To be used to process the annotated Java classes and populate the RuntimeMetadata model.  
+	 */
+	AnnotationProcessor processor;
+	
+	/**
+	 * To be used to create knowledgeManager objects. 
+	 */
+	KnowledgeManagerFactory knowledgeManagerFactory;
+	
+	/**
+	 * Contains the initialized plugins for this deeco object.
+	 */
+	Map<Class<? extends DEECoPlugin>, DEECoPlugin> pluginsMap;
+	
+	/**
+	 * Maintains the set of known ensemble definitions for the purpose of guarding against duplicate deployment of one ensemble.
+	 */
+	Set<Class> knownEnsembleDefinitions;
```

Comment:
This makes a lot of sense. But shouldn't it be rather part of the AnnotationProcessor (including the check which is currently in the deployEnsemble)?